### PR TITLE
Add PR review agent

### PR DESCRIPTION
**Pull request by mehtarac**
## Description

This PR adds an Agent as a github action workflow. This agent can be used following a `/strands review` command, which triggers the action to start the review process. During the review:
1. Agent sets the the review environment reading the PR description, current branch, and identifies the type of change.
2. Identifies the linked issue, checks for existing review comments and uses a new tool, `get_pr_files` to review changed files and scope. 
3. Reviews repo guidelines from CONTRIBUTING.md and AGENTS.md
4. Does a code analysis for
  - File and directory strucuture, naming conventions, and breaking changes
  - Language best practices
  - Testing review
  - Documentation review
5. Generates review comments
6. Posts review comments
## Related Issues

[#1](https://github.com/strands-agents/private-sdk-typescript-staging/issues/1)

## Type of Change

New feature

## Testing

How have you tested the change?
Tested in forked repo on a sample PR: https://github.com/mehtarac/sdk-typescript/pull/1

- [x] I ran `npm run check`

## Checklist
- [x] I have read the CONTRIBUTING document
- [x] I have added any necessary tests that prove my fix is effective or my feature works
- [x] I have updated the documentation accordingly
- [x] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/agent-sops/task-reviewer.sop.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+- You MUST check for proper separation of concerns
+- You MUST note any violations of the composability tenet
+
+#### 4.2 Code Quality Review
+
+Examine the code for quality, readability, and maintainability issues.
+
+**Constraints:**
+- You MUST check for TypeScript best practices:
+  - No `any` types
```
Comment (Unshure) [inline .github/agent-sops/task-reviewer.sop.md]: Lets not make this typescript specific, lets make it generic so we can copy it to all of our repos

File: .github/agent-sops/task-reviewer.sop.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+#### 4.3 Testing Review
+
+Analyze the test coverage and quality of tests.
+
+**Constraints:**
+- You MUST verify that new functionality has corresponding tests
+- You MUST check that tests follow the patterns in `docs/TESTING.md` if available
+- You MUST ensure tests are in the correct directories (`src/**/__tests__/**` for unit tests)
+- You MUST verify test coverage meets the 80% requirement
```
Comment (Unshure) [inline .github/agent-sops/task-reviewer.sop.md]: I dont think the pr reviewer should care about code coverage. This is deterministic, and can be covered outside of the agent.

File: .github/agent-sops/task-reviewer.sop.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+- You MUST identify missing edge cases or error scenarios
+- You MUST verify integration tests are included when appropriate
+
+#### 4.4 Documentation Review
+
+Check documentation completeness and quality.
+
+**Constraints:**
+- You MUST verify TSDoc comments exist for all exported functions
+- You MUST check that documentation is clear and helpful
```
Comment (Unshure) [inline .github/agent-sops/task-reviewer.sop.md]: clear, helpful, AND CONCISE!

File: .github/agent-sops/task-reviewer.sop.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+Format review comments to be clear and actionable.
+
+**Constraints:**
+- You MUST start with a clear summary of the issue
+- You MUST explain why the change is needed
+- You MUST provide a specific suggestion or solution
+- You MUST include code examples when helpful
+- You MUST reference documentation or standards when applicable
+- You SHOULD use this format:
+  ```
```
Comment (Unshure) [inline .github/agent-sops/task-reviewer.sop.md]: Im worried this might lead to very verbose feedback. Let reduce the example, and tell it to be concise

File: .github/agent-sops/task-reviewer.sop.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+- You MUST verify that new changes don't introduce other issues
+- You MUST acknowledge when feedback has been properly addressed
+- You MUST provide approval when all critical issues are resolved
+- You SHOULD be responsive to questions from the author
+
+## Review Focus Areas
+
+### Code Quality Priorities
+
+1. **Type Safety**: Ensure proper TypeScript usage without `any` types
```
Comment (Unshure) [inline .github/agent-sops/task-reviewer.sop.md]: nit: Make generic, not typescript specific

File: .github/agent-sops/task-reviewer.sop.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+interface UserData {
+  id: string;
+  name: string;
+  email: string;
+}
+
+function processUser(userData: UserData): void {
+  // Implementation
+}
+```
```
Comment (Unshure) [inline .github/agent-sops/task-reviewer.sop.md]: Lets not give code examples since this should be language agnostic

File: .github/agent-sops/task-reviewer.sop.md (hunk 1 above)
Comment (mehtarac) [reply .github/agent-sops/task-reviewer.sop.md]: Updated

File: .github/agent-sops/task-reviewer.sop.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+
+### 6. Post Review Comments
+
+Add the review comments to the pull request.
+
+**Constraints:**
+- You MUST post comments on specific lines where issues are identified
+- You MUST use the `add_pr_inline_comment` tool for line-specific feedback on code
+- You MUST use the `reply_to_review_comment` tool for line-specific feedback
+- You MUST use the `add_issue_comment` tool for general PR-level comments
```
[the remark this answers is not in the document]
Comment (mehtarac) [reply .github/agent-sops/task-reviewer.sop.md]: modified the sop to update with the suggested instructions

File: .github/scripts/python/github_tools.py [hunk 8]
```diff
@@ -450,6 +450,114 @@ def reply_to_review_comment(pr_number: int, comment_id: int, reply_text: str, re
...
+        pr_number: The pull request number
+        body: The comment text
+        path: The file path to comment on (optional; if omitted, creates general PR comment)
+        line: The line number to comment on (optional; if omitted with path, creates file-level comment)
+        repo: GitHub repository in the format "owner/repo" (optional; falls back to env var)
+
+    Returns:
+        Result of the operation
+    """
+    # If no path provided, create a general PR comment (issue comment)
```
Comment (Unshure) [inline .github/scripts/python/github_tools.py]: What is a general pr comment? Is this the same as issue_comment? Im not against having two tools that do the same thing, just trying to figure this out.

File: .github/agent-sops/task-reviewer.sop.md [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+### 2. Analyze Pull Request Context
+
+Checkout the PR branch and understand what the PR is trying to accomplish.
+
+**Constraints:**
+- You MUST checkout the PR branch to review the actual changes
+- You MUST read the pull request description and understand the purpose of the changes
+- You MUST note the PR number and branch name in your notebook
+- You MUST identify the type of changes (feature, bugfix, refactor, etc.)
+- You MUST read the PR description thoroughly
```
Comment (Unshure) [inline .github/agent-sops/task-reviewer.sop.md]: Nit: this is a repeat

**Review by Unshure — DISMISSED** [review]
Lets run the pr review sop to review this pr :D
